src/utils/fees.ts

```diff
@@ -14,14 +14,15 @@ export const calculateFee = (gas: Gas): BigNumber => {
 };
 
 export const calculateTransactionResponseFee = (
-  transactionResponse: TransactionResponse
+  transactionResponse: TypeSafeTransactionResponse
 ): BigNumber | undefined => {
   const { gasLimit, gasPrice, maxFeePerGas } = transactionResponse;
-  const gas: Gas | undefined = maxFeePerGas
-    ? { data: { gasLimit, maxFeePerGas }, type: "eip-1559" }
-    : gasPrice
-    ? { data: { gasLimit, gasPrice }, type: "legacy" }
-    : undefined;
+  const gas: Gas | undefined =
+    gasLimit && maxFeePerGas
```

Comment:
For example, we shouldn't check the `gasLimit` prop here as it will always be there, but we are forced to do it because of the new `TypeSafeTransactionResponse` type.

Reply:
Yep, that's precisely the point!